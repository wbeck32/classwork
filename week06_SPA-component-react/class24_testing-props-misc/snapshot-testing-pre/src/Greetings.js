import React, { Component } from 'react';

export function Greeter({ salutation, name }) {
  return <div>{salutation || 'Hello'} {name}</div>;
}

export class Greetings extends Component {
  constructor() {
    super();
    this.state = {
      salutation: 'Hello'
    };
  }

  changeSalutation() { 
    this.setState({
      salutation: this.getOppositeSalutation()
    });
  }

  getOppositeSalutation() {
    return this.state.salutation === 'Hello' ? 'Goodbye' : 'Hello'
  }

  render() {
    const { salutation } = this.state;
    const { people } = this.props;
    return (
      <div>
        <ul>
          {people.map((person, i) => (
            <Greeter key={i} salutation={salutation} name={person} />
          ))}
        </ul>
        <button onClick={() => this.changeSalutation()}>
          Say {this.getOppositeSalutation()}
        </button>
      </div>
    );
  }
}
