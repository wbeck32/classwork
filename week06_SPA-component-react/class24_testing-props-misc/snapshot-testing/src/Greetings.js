import React, { Component } from 'react';
import PropTypes from 'prop-types';

Greeter.propTypes = {
  salutation: PropTypes.string,
  name: PropTypes.string.isRequired
}

export function Greeter({ salutation = 'Hello', name }) {
  console.log('Greeter', salutation, name);
  return <div>{salutation} {name}</div>;
}

export class Greetings extends Component {

  static propTypes = {
    people: PropTypes.array.isRequired
  }

  constructor() {
    super();
    this.state = {
      salutation: 'Hello'
    };
  }

  changeSalutation(salutation) { 
    this.setState({
      salutation: salutation || this.getOppositeSalutation()
    });
  }

  getOppositeSalutation() {
    return this.state.salutation === 'Hello' ? 'Goodbye' : 'Hello'
  }

  render() {
    const { salutation } = this.state;
    const { people } = this.props;
    return (
      <div>
        <ul>
          {people.map((person, i) => (
            <Greeter key={i} salutation={salutation} name={person} />
          ))}
        </ul>
        <button onClick={() => this.changeSalutation()}>
          Say {this.getOppositeSalutation()}
        </button>
        <input onChange={({ target }) => this.changeSalutation(target.value)}/>
      </div>
    );
  }
}
