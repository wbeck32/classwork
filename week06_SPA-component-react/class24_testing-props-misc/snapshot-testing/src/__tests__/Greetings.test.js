import React from 'react';
// "mount" does full render including children
// "shallow" does just the parent
import { shallow /*, mount*/ } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Greeter, Greetings } from '../Greetings';

describe('Greeter', () => {

    it('says salutation to Mimi', () => {
        const wrapper = shallow(<Greeter salutation="Goodbye" name="Mimi" />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('says hello by default', () => {
        const wrapper = shallow(<Greeter name="Mimi" />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

});


describe('Greetings', () => {

    it('says hello to people', () => {
        const wrapper = shallow(
            <Greetings people={['Jill', 'Julie', 'Jonsie']} />
        );
        // should default to Hello
        expect(toJSON(wrapper)).toMatchSnapshot();
        
        // // direct call to component API:        
        // const component = wrapper.instance();
        // component.changeSalutation();
        // wrapper.update();

        // or simulate the event:
        const button = wrapper.find('button');
        button.simulate('click');
        
        // should switch to opposite "Goodbye"
        expect(toJSON(wrapper)).toMatchSnapshot();
        
        const input = wrapper.find('input');
        input.simulate('change', { target: { value: 'foo' }});
        
        // should be 'foo', the value from the input change
        expect(toJSON(wrapper)).toMatchSnapshot();
        
        // should revert to "Hello"
        button.simulate('click');
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});    
