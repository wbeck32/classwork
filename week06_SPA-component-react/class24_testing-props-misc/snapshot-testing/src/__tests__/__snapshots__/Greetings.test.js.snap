// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeter says hello by default 1`] = `
<div>
  Hello
   
  Mimi
</div>
`;

exports[`Greeter says salutation to Mimi 1`] = `
<div>
  Goodbye
   
  Mimi
</div>
`;

exports[`Greetings says hello to people 1`] = `
<div>
  <ul>
    <Greeter
      name="Jill"
      salutation="Hello"
    />
    <Greeter
      name="Julie"
      salutation="Hello"
    />
    <Greeter
      name="Jonsie"
      salutation="Hello"
    />
  </ul>
  <button
    onClick={[Function]}
  >
    Say 
    Goodbye
  </button>
  <input
    onChange={[Function]}
  />
</div>
`;

exports[`Greetings says hello to people 2`] = `
<div>
  <ul>
    <Greeter
      name="Jill"
      salutation="Goodbye"
    />
    <Greeter
      name="Julie"
      salutation="Goodbye"
    />
    <Greeter
      name="Jonsie"
      salutation="Goodbye"
    />
  </ul>
  <button
    onClick={[Function]}
  >
    Say 
    Hello
  </button>
  <input
    onChange={[Function]}
  />
</div>
`;

exports[`Greetings says hello to people 3`] = `
<div>
  <ul>
    <Greeter
      name="Jill"
      salutation="foo"
    />
    <Greeter
      name="Julie"
      salutation="foo"
    />
    <Greeter
      name="Jonsie"
      salutation="foo"
    />
  </ul>
  <button
    onClick={[Function]}
  >
    Say 
    Hello
  </button>
  <input
    onChange={[Function]}
  />
</div>
`;

exports[`Greetings says hello to people 4`] = `
<div>
  <ul>
    <Greeter
      name="Jill"
      salutation="Hello"
    />
    <Greeter
      name="Julie"
      salutation="Hello"
    />
    <Greeter
      name="Jonsie"
      salutation="Hello"
    />
  </ul>
  <button
    onClick={[Function]}
  >
    Say 
    Goodbye
  </button>
  <input
    onChange={[Function]}
  />
</div>
`;
